Source/Core/Core/FifoPlayer/FifoRecordAnalyzer.cpp

```diff
@@ -44,14 +45,28 @@ void FifoRecordAnalyzer::WriteVertexArray(int arrayIndex, const u8* vertexData,
                                           int numVertices)
 {
   // Skip if not indexed array
-  int arrayType = (s_CpMem.vtxDesc.Hex >> (9 + (arrayIndex * 2))) & 3;
-  if (arrayType < 2)
+  VertexComponentFormat arrayType;
+  if (arrayIndex == 0)
+    arrayType = s_CpMem.vtxDesc.low.Position;
+  else if (arrayIndex == 1)
+    arrayType = s_CpMem.vtxDesc.low.Normal;
+  else if (arrayIndex == 2 || arrayIndex == 3)
+    arrayType = s_CpMem.vtxDesc.low.Color[arrayIndex - 2];
+  else if (arrayIndex < 12)
+    arrayType = s_CpMem.vtxDesc.high.TexCoord[arrayIndex - 4];
+  else
+  {
+    PanicAlertFmt("Invalid arrayType {}", arrayType);
+    return;
+  }
```

Comment:
The PanicAlert should use arrayIndex, and also the `ARRAY_POSITION`/etc constants should be used here.